### PR TITLE
feat(image): add decorative prop to component - FE-5362

### LINKED ISSUE #5469: Add / Clarify Image component behaviour
### Desired behaviour

As requested: raising a GitHub request for the following two changes:
- alt prop will be clarified to always be provided
- ability to set aria-hidden

> Ah I see and thanks 
[@Robin Zigmond](https://sageone.slack.com/team/UUADCMQHH), perhaps to prevent the abuse of passing of null alt strings, we could add a new prop to Image such as decorative which would automatically handle cases like this.
Please could you add this as an enhancement on our github repo 
[@Paul](https://sageone.slack.com/team/ULVLHRX2Q)
?

### Current behaviour

An error is given if alt prop is empty string and there is no way to set aria hidden

### Suggested Solution

n/a

### CodeSandbox or Storybook URL

_No response_

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by DipperTheDan**
Introduces a decorative prop to the Image component so that alt text can be an empty string without triggering an invariant.

fixes #5469

### Proposed behaviour

Add a `decorative` prop to the `Image` component so that a decorative image can be displayed without triggering an invariant. This means that `src` with `alt=""` is now supported as long as the `decorative` prop is passed.

### Current behaviour

Having a `src` and  `alt=""` triggers an invariant warning.

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- Added a new story for the `decorative` prop so make sure that story works as expected. 
- Can also be tested using this codesandbox:
https://codesandbox.io/s/epic-darkness-czg4z9

### CODE REVIEW

File: src/components/image/image.stories.mdx [hunk 1]
```diff
@@ -58,6 +58,17 @@ To hide the component completely - meaning hidden visually and from being access
   <Story name="with hidden prop" story={stories.WithHiddenProp} />
 </Canvas>
 
+### Decorative Image
+
+There may be a use case where a decorative image is required. A decorative images does not add information to the content of a page.
```
Comment (Parsium) [inline src/components/image/image.stories.mdx:63]:
```suggestion
There may be a use case where a decorative image is required. A decorative images do not add information to the content of a page. 
```

Comment (DipperTheDan) [reply src/components/image/image.stories.mdx:63]: Probably should be:
`Decorative images do not add...`

**Review by Parsium — APPROVED** [review]

File: src/components/image/image.component.tsx [hunk 2]
```diff
@@ -2,19 +2,25 @@ import React from "react";
...
+}: StyledImageProps) => {
   invariant(
     !src || !children,
     "The 'Image' component renders as an 'img' element when the 'src' prop is used and therefore does not accept children."
   );
 
   invariant(
-    !src || (alt && typeof alt === "string"),
-    "Please provide an 'alt' string when rendering the 'Image' component as an 'img' element."
+    !src || (alt && typeof alt === "string") || decorative,
```
Comment (edleeks87) [inline src/components/image/image.component.tsx:18]: I realise you didn't add this, but why are we checking the `typeof` the `alt` prop as well when the interface says it's either `undefined | string`?

Comment (edleeks87) [reply src/components/image/image.component.tsx:18]: Happy to approve for now as it's not breaking anything

**Review by edleeks87 — APPROVED** [review]

File: src/components/image/image.component.tsx (hunk 2 above)
Comment (DipperTheDan) [reply src/components/image/image.component.tsx:18]: Cheers. I did look at this and think about changing it but getting the work done to make sure Nebula were not blocked took priority. I'll get something on the backlog so we don't forget to look at this in the near future 😄

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**